File: src/AnalyticsService/Service/Model/GeneralBatchStats.fs
namespace AnalyticsService.Service.Model

open System

/// A record representing a general statistic about a document batch.
type GeneralBatchStats =
  { AvgNumberOfDocs: int
    SuccessRate: float
    AvgDuration: TimeSpan }
    static member Default = { AvgNumberOfDocs = 0; SuccessRate = 0; AvgDuration = TimeSpan.Zero }

File: src/AnalyticsService/Service/Model/GeneralBatchStats.cs
namespace AnalyticsService.Service.Model;

/// <summary>
/// A record representing a general statistic about a document batch.
/// </summary>
internal sealed record GeneralBatchStat(
    int AvgNumberOfDocuments,
    float SuccessRate,
    TimeSpan AvgDuration
)
{
    public static GeneralBatchStat Default => new(0, 0, TimeSpan.Zero);
};